keyboards/illusion/rosa/rosa.c

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2021 Brandon Lee
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "rosa.h"
+
+void keyboard_pre_init_kb(void) {
+    setPinOutput(B3); // caps lock indicator LED pin
+}
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if (res) {
+        writePin(B3, !led_state.caps_lock);
+    }
+    return res;
+}
```

Comment:
Can be replaced with a simple define in config.h: https://docs.qmk.fm/#/feature_led_indicators

Reply:
I've never done LED indicators in config.h like this, but I've made what I think are the appropriate changes: 02cd17b2b8a9203885e12aceb260237cce27b287